pkg/controller/statusmanager/machineconfig_watcher.go

```diff
@@ -0,0 +1,88 @@
+package statusmanager
+
+import (
+	"context"
+
+	configv1client "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned"
+	mcfgv1informer "github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/machineconfiguration.openshift.io/v1"
+	mcfgv1lister "github.com/openshift/machine-config-operator/pkg/generated/listers/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type MachineConfigsWatcher struct {
+	status *StatusManager
+}
+
+func (s *StatusManager) initInformerForMachineConfigs(cfg *rest.Config) error {
+	configClient, err := configv1client.NewForConfig(cfg)
```

Comment:
If you use a new client, you won't share the cache with the manager. You wouldn't have this problem with the kind source since it forces you to think about what cache you are using, However if you want to use infromers anyway, use the same client used for the manager.